moveit_setup_assistant/src/widgets/collision_matrix_model.cpp

```diff
@@ -0,0 +1,212 @@
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Robert Haschke */
+
+#include "collision_matrix_model.h"
+#include "default_collisions_widget.h"
+#include <boost/unordered_map.hpp>
+#include <boost/assign.hpp>
+#include <QVector>
+
+using namespace moveit_setup_assistant;
+
+/// Boost mapping of reasons for disabling a link pair to strings
+static const boost::unordered_map<moveit_setup_assistant::DisabledReason, const char *> longReasonsToString =
+    boost::assign::map_list_of  // clang-format off
+    ( moveit_setup_assistant::NEVER, "Never in Collision" )
+    ( moveit_setup_assistant::DEFAULT, "Collision by Default" )
+    ( moveit_setup_assistant::ADJACENT, "Adjacent Links" )
+    ( moveit_setup_assistant::ALWAYS, "Always in Collision" )
+    ( moveit_setup_assistant::USER, "User Disabled" )
+    ( moveit_setup_assistant::NOT_DISABLED, "");  // clang-format on
+
+/// Boost mapping of reasons to a background color
+static const boost::unordered_map<moveit_setup_assistant::DisabledReason, QVariant> longReasonsToBrush =
+    boost::assign::map_list_of  // clang-format off
+    ( moveit_setup_assistant::NEVER, QBrush(QColor("lightgreen")) )
+    ( moveit_setup_assistant::DEFAULT, QBrush(QColor("lightpink")) )
+    ( moveit_setup_assistant::ADJACENT, QBrush(QColor("powderblue")) )
+    ( moveit_setup_assistant::ALWAYS, QBrush(QColor("tomato")) )
+    ( moveit_setup_assistant::USER, QBrush(QColor("yellow")) )
+    ( moveit_setup_assistant::NOT_DISABLED, QBrush());  // clang-format on
```

Comment:
This would best be documented http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/setup_assistant/setup_assistant_tutorial.html